### PR TITLE
broadcast arguments in jax.numpy.take_along_axis

### LINKED ISSUE #1521: np.take_along_axis gives wrong gradient
The following repro script shows the issue:
```
import jax.numpy as np
from jax import value_and_grad

idx = np.repeat(np.arange(3), 10).reshape((30, 1))

def f(x):
    y = x * np.arange(3.).reshape((1, 3))
    return np.take_along_axis(y, idx, -1).sum()

def g(x):
    y = x * np.arange(3.).reshape((1, 3))
    y = np.broadcast_to(y, (30, 3))
    return np.take_along_axis(y, idx, -1).sum()

print(value_and_grad(f)(1.))  # get 30, 0
print(value_and_grad(g)(1.))  # get 30, 30
```

I think `(30, 30)` is the correct answer.

### PR DESCRIPTION
**Pull request by mattjj**
fixes #1521

The gather failure in the first test I wrote was misguided because I think it was doing out-of-bounds accessing. It was just a bad repro.

After reading #1521 more carefully, I realized it was just a broadcasting issue in our implementation of `take_along_axis`.

### CODE REVIEW

**Comment by mattjj** [comment]
Ah, we need to do the broadcasting on everything except the axis along which we're indexing IIUC...

**Comment by fehiepsi** [comment]
Yes, that’s what I did to fix that issue in numpyro.
